spec/intl.html

```diff
@@ -0,0 +1,372 @@
+            1. Throw a *RangeError* exception.
+          1. Let _timeZone_ be CanonicalizeTimeZoneName(_timeZone_).
+        1. Else,
+          1. Let _timeZone_ be DefaultTimeZone().
+        1. Set _dateTimeFormat_.[[TimeZone]] to _timeZone_.
+        1. Let _opt_ be a new Record.
+        1. For each row of <emu-xref href="#table-datetimeformat-components"></emu-xref>, except the header row, in table order, do
+          1. Let _prop_ be the name given in the Property column of the row.
+          1. Let _value_ be ? GetOption(_options_, _prop_, `"string"`, &laquo; the strings given in the Values column of the row &raquo;, *undefined*).
+          1. Set _opt_.[[&lt;_prop_&gt;]] to _value_.
+        1. Let _dataLocaleData_ be _localeData_.[[&lt;_dataLocale_&gt;]].
+        1. Let _formats_ be _dataLocaleData_.[[formats]].
+        1. Let _matcher_ be ? GetOption(_options_, `"formatMatcher"`, `"string"`, &laquo; `"basic"`, `"best fit"` &raquo;, `"best fit"`).
+        1. If _matcher_ is `"basic"`, then
+          1. Let _bestFormat_ be BasicFormatMatcher(_opt_, _formats_).
+        1. Else,
+          1. Let _bestFormat_ be BestFitFormatMatcher(_opt_, _formats_).
+        1. For each row in <emu-xref href="#table-datetimeformat-components"></emu-xref>, except the header row, in table order, do
+          1. Let _prop_ be the name given in the Property column of the row.
+          1. Let _p_ be _bestFormat_.[[&lt;_prop_&gt;]].
+          1. If _p_ not *undefined*, then
+            1. Set _dateTimeFormat_'s internal slot whose name is the Internal Slot column of the row to _p_.
+        1. If _dateTimeFormat_.[[Hour]] is not *undefined*, then
+          1. Let _hcDefault_ be _dataLocaleData_.[[hourCycle]].
+          1. Let _hc_ be _dateTimeFormat_.[[HourCycle]].
+          1. If _hc_ is *null*, then
+            1. Set _hc_ to _hcDefault_.
+          1. If _hour12_ is not *undefined*, then
+            1. If _hour12_ is *true*, then
+              1. If _hcDefault_ is `"h11"` or `"h23"`, then
+                1. Set _hc_ to `"h11"`.
+              1. Else,
+                1. Set _hc_ to `"h12"`.
+            1. Else,
+              1. Assert: _hour12_ is *false*.
+              1. If _hcDefault_ is `"h11"` or `"h23"`, then
+                1. Set _hc_ to `"h23"`.
+              1. Else,
+                1. Set _hc_ to `"h24"`.
+          1. Set _dateTimeFormat_.[[HourCycle]] to _hc_.
+          1. If _dateTimeformat_.[[HourCycle]] is `"h11"` or `"h12"`, then
+            1. Let _pattern_ be _bestFormat_.[[pattern12]].
+          1. Else,
+            1. Let _pattern_ be _bestFormat_.[[pattern]].
+        1. Else,
+          1. Set _dateTimeFormat_.[[HourCycle]] to *undefined*.
+          1. Let _pattern_ be _bestFormat_.[[pattern]].
+        1. Set _dateTimeFormat_.[[Pattern]] to _pattern_.
+        1. <ins>For each row in <emu-xref href="#table-temporal-patterns"></emu-xref>, except the header row, in table order, do</ins>
```

Comment:
How about making a new abstract operation `GetDateTimeFormatPattern(limitedOptions, formats)` that runs this logic but also the logic for the regular non-Temporal case (in which limitedOptions == options)?  I did something similar in Intl.NumberFormat with a new abstract operation `GetNumberFormatPattern`.